### PR TITLE
Add templates, and remove relationships (playground)

### PR DESCRIPTION
**Pull request by andreavs**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,15 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [0.42.0] - 21-02-01
+
+### Added 
+- Alpha CogniteClient.
```
Comment (sanderland) [inline CHANGELOG.md:18]: no

File: cognite/experimental/data_classes/templates.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+from cognite.client.data_classes._base import *
+
+
+class TemplateGroup(CogniteResource):
+    """A template group is a high level concept encapsulating a schema and a set of template instances. It also has query capability support.
+
+    Template groups are versioned, so there can be multiple template groups with the same external ID.
+    The versioning is happening automatically whenever a template groups is changed.
+
+    GraphQL SDL is used as the language to describe the structure of the templates and data types.
```
Comment (sanderland) [inline cognite/experimental/data_classes/templates.py:12]: The GraphQL schema definition language

File: cognite/experimental/data_classes/templates.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+class TimeSeriesResolver(CogniteResource):
+    """Resolves a field of type 'TimeSeries' to a specific Time Series.
+
+    Args:
+        time_series_id (str): The time series External ID.
+    """
+
+    def __init__(self, time_series_id: str = None, cognite_client=None):
+        self.type = "timeSeries"
+        self.time_series_id = time_series_id
```
Comment (sanderland) [inline cognite/experimental/data_classes/templates.py:111]: .id ?

**Review by sanderland — CHANGES_REQUESTED** [review]
pizza time, will have another look tomorrow.

File: cognite/experimental/data_classes/templates.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    ):
+        self.type = "raw"
+        self.db_name = db_name
+        self.table_name = table_name
+        self.row_key = row_key
+        self.column_name = column_name
+        self._cognite_client = cognite_client
+
+
+class TimeSeriesResolver(CogniteResource):
```
Comment (Tapped) [inline cognite/experimental/data_classes/templates.py:102]: Actually the TimeSeriesResolver is deprecated. Instead the `ConstantResolver` or `RawResolver` should be used to retrieve the externalId of the TimeSeries.

Comment (sanderland) [reply cognite/experimental/data_classes/templates.py:102]: NB this breaks the template completion in a way that's not clear how to resolve.

File: cognite/experimental/_api/templates.py [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+from cognite.experimental.data_classes.templates import *
+
+
+class TemplatesAPI(APIClient):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.groups = TemplateGroupsApi(*args, **kwargs)
+        self.versions = TemplateGroupVersionsApi(*args, **kwargs)
+        self.instances = TemplateInstancesApi(*args, **kwargs)
+        self.query = TemplatesQuery(*args, **kwargs)
```
Comment (sanderland) [inline cognite/experimental/_api/templates.py:12]: can this be simpler? e.g. i see no reason for .query to be in a sub-api.

Comment (sanderland) [reply cognite/experimental/_api/templates.py:12]: my intuition is to have instances+query live in the top level, but you may understand templated better

**Review by sanderland — CHANGES_REQUESTED** [review]
some more comments.
+ docs in cognite.rst please

File: cognite/experimental/data_classes/templates.py (hunk 4 above)
Comment (Tapped) [reply cognite/experimental/data_classes/templates.py:102]: Yes, but we'll have to provide you an API for doing it. In the meanwhile assuming that a TimeSeries type is a ConstantResolver will work.

File: cognite/experimental/_api/templates.py (hunk 5 above)
Comment (andreavs) [reply cognite/experimental/_api/templates.py]: Tried to roughly follow how the 3d sdk is designed, which also deals with wanting to flatten a structure that is heavily nested in the api. In that case, e.g. the asset mappings are in their own sub-api, even though the path in the api is under the model revisions. That being said, i do not have strong feelings on it, so we can move it

File: cognite/experimental/data_classes/templates.py [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+            else TemplateInstance._encode_field_resolvers(value, camel_case=camel_case)
+            for key, value in self.__dict__.items()
+            if value not in EXCLUDE_VALUE and not key.startswith("_")
+        }
+
+    def _encode_field_resolvers(field_resolvers: Dict[str, FieldResolvers], camel_case):
+        return {key: value.dump(camel_case=camel_case) for key, value in field_resolvers.items()}
+
+    @classmethod
+    def _load(cls, resource: Union[Dict, str], cognite_client=None):
```
Comment (sanderland) [inline cognite/experimental/data_classes/templates.py:195]: can this be .super() into fixing fieldresolvers?

File: docs/source/cognite.rst [hunk 7]
```diff
@@ -497,3 +466,67 @@ Data classes
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. automethod:: cognite.experimental._api.templates.TemplateGroupVersionsAPI.list
+
+Delete a Temple Group version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. automethod:: cognite.experimental._api.templates.TemplateGroupVersionsAPI.delete
+
+Run a GraphQL query
+^^^^^^^^^^^^^^^^^^^
+.. automethod:: cognite.experimental._api.templates.TemplateGroupVersionsAPI.graphql_query
```
Comment (sanderland) [inline docs/source/cognite.rst:507]: AttributeError: graphql_query

**Review by sanderland — CHANGES_REQUESTED** [review]
docs fix please, make html fails.

**Review by sanderland — APPROVED** [review]

**Pull request merged**